src/pollingManager.ts

```diff
@@ -41,23 +50,61 @@ export class PollingManager {
     }
 
     this.logger.debug({ ...roiJobs, msg: `Handling ROI jobs` });
-    if (roiJobs.completedJobs?.length != null) {
+    if (roiJobs.completedJobs && roiJobs.completedJobs.length > 0) {
       existsJobs = true;
       this.logger.debug({ msg: `ROI Completed jobs detected, running finalize job` });
       for (const job of roiJobs.completedJobs) {
-        this.logger.info({ jobId: job.id, msg: `Execute completed job finalizing on ROI exporting for job: ${job.id}` });
-        await this.taskManager.finalizeExportJob(job, expirationDate);
+        this.logger.info({ jobId: job.id, msg: `Execute completed job finalizing task creation on ROI exporting for job: ${job.id}` });
+        await this.taskManager.createFinalizeTask(job, this.finalizeTaskType);
       }
-    } else if (roiJobs.failedJobs?.length != null) {
+    } else if (roiJobs.failedJobs && roiJobs.failedJobs.length > 0) {
       existsJobs = true;
       this.logger.debug({ msg: `ROI Failed jobs detected, running finalize job` });
       for (const job of roiJobs.failedJobs) {
-        this.logger.info({ jobId: job.id, msg: `Execute failed job finalizing on ROI exporting for job: ${job.id}` });
+        this.logger.info({ jobId: job.id, msg: `Execute failed job finalizing task creation on ROI exporting for job: ${job.id}` });
         const gpkgFailedErr = `failed to create gpkg, job: ${job.id}`;
-        await this.taskManager.finalizeExportJob(job, expirationDate, false, gpkgFailedErr);
+        await this.taskManager.createFinalizeTask(job, this.finalizeTaskType, false, gpkgFailedErr);
       }
     }
 
     return existsJobs;
   }
+
+  public async jobFinalizePoll(): Promise<boolean> {
+    const expirationDate = new Date();
+    expirationDate.setDate(expirationDate.getDate() + this.expirationDays);
+
+    const finalizeTask = await this.queueClient.queueHandlerForFinalizeTasks.dequeue<ITaskFinalizeParameters>(this.finalizeTaskType);
+    if (finalizeTask) {
+      const attempts = finalizeTask.attempts;
+      const jobId = finalizeTask.jobId;
+      const taskId = finalizeTask.id;
+      try {
+        this.logger.info({ jobId: finalizeTask.jobId, taskId: finalizeTask.id, msg: `Found new finalize task` });
+        const job = (await this.taskManager.getFinalizeJobById(finalizeTask.jobId)) as JobFinalizeResponse;
+
+        if (attempts <= this.finalizeAttempts) {
+          const isSuccess = finalizeTask.parameters.exporterTaskStatus === OperationStatus.COMPLETED ? true : false;
+          const errReason = finalizeTask.parameters.reason;
+          await this.taskManager.finalizeExportJob(job, expirationDate, isSuccess, errReason);
```

Comment:
the following:
> if (// GPKG task finished with success) { 
  finalizeGPKGSuccess
} 
finalizeGPKGFailure()..

then

> finalizeGPKGSuccess() {...
> bool metadataCreatedSuccessfully = await this.taskManager.createMetadata(job, expirationDate, isSuccess, errReason);
> await this.taskManager.invokeWebhooks();
> if (metadataCreatedSuccessfully) {
   // close task success
   // close job success
} else {
   // close task failed
   // close job failed
}
...}

> finalizeGPKGFailure() {...
await this.taskManager.invokeWebhooks();
// ack()
// fail job
// deleteTask() // after job failure in order to prevent polling to recreate the finalization
...}